udtube/data/conllu.py

```diff
@@ -23,19 +24,21 @@
 ]
 
 
-class TokenList:
+class TokenList(collections.UserList):
     """TokenList object.
 
+    This behaves like a list of tokens (of type Dict[str, str]) with
+    optional associated metadata.
+
     Args:
         tokens: List of tokens.
         metadata: ordered dictionary of string/key pairs.
     """
 
-    tokens: List[Dict[str, str]]
     metadata: Dict[str, Optional[str]]
 
-    def __init__(self, tokens: List[Dict[str, str]], metadata=None):
-        self.tokens = tokens
+    def __init__(self, tokens: Iterable[Dict[str, str]], metadata=None):
```

Comment:
The instance variable `self.metadata` is type-hinted on line 38, so the argument `metadata` doesn't need to be. This seems to be the standard: hint the instance variables rather than the constructor args. `mypy` and friends (which we haven't enabled yet) usually can reason, statically, and backwards, from this, and if we're wrong I can always add the hint to the constructor arg later.

Reply:
SGTM, thanks